comment_scan.rb

```diff
@@ -85,24 +85,23 @@ def restart(num_to_post, bundle)
             end
           when 'huh?'
             matched_regexes = report_raw(comment["post_type"], comment["body_markdown"])
-            reason_text = ""
-            #Go through regexes we matched to build reason_text
-            matched_regexes.each { |regex_matched| reason_text += "Matched reason \"#{Reason.where(id: regex_matched["reason_id"])[0]["name"]}\" for regex: #{regex_matched["regex"]}\n" }
-            #If post isn't deleted, check if this was an inactive comment
-            if !isPostDeleted(cli, comment["post_id"])
-              post = Array(cli.posts(comment["post_id"].to_i))[0]
-              if Time.at(post.json["last_activity_date"].to_i).to_date < Time.at(comment["creation_date"].to_i).to_date - 30
-                reason_text += "Comment was made #{(Time.at(comment["creation_date"].to_i).to_date - Time.at(post.json["last_activity_date"].to_i).to_date).to_i} days after last activity on post\n"
+            # Go through regexes we matched to build reason_text
+            reason_text = matched_regexes.map { |rm| "Matched reason \"#{Reason.where(id: rm["reason_id"]).first.name}\" for regex: #{rm.regex}" }.join("\n")
+            # If post isn't deleted, check if this was an inactive comment
+            if !post_deleted?(cli, comment["post_id"])
+              post = cli.posts(comment["post_id"].to_i).first
+              if timestamp_to_date(post.json["last_activity_date"]) < timestamp_to_date(comment["creation_date"]) - 30
+                reason_text += "Comment was made #{(timestamp_to_date(comment["creation_date"]) - timestamp_to_date(post.json["last_activity_date"])).to_i} days after last activity on post\n"
               end
             end
-            #Check if we're high on toxicity
+            # Check if we're high on toxicity
             if comment["toxicity"].to_f >= 0.7
               reason_text += "Comment has toxicity of #{comment["toxicity"]}\n"
             end
             
-            reason_text = reason_text.chomp #chompt to eat that last newline
+            reason_text = reason_text.chomp # chomp to eat that last newline
             if !reason_text.to_s.empty?
```

Comment:
This could prolly be done with a ternary operator: `cb.say (reason_text.empty? ? "Comment didn't match any regexes" : reason_text) room_id`